### ISSUE TITLE
ArrayIndexOutOfBoundsException during startup with mpOpenApi

### ISSUE THREAD

**Issue by Azquelt** [OpenLiberty/open-liberty#19589]
**Describe the bug**  
During startup, mpOpenApi uses jandex to index classes and the following exception was observed in IBM/FHIR#3092

```
java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at org.jboss.jandex.Indexer.updateTypeTarget(Indexer.java:1120)
	at org.jboss.jandex.Indexer.updateTypeTargets(Indexer.java:842)
	at org.jboss.jandex.Indexer.index(Indexer.java:1970)
	at io.openliberty.microprofile.openapi20.utils.IndexUtils.indexContainer(IndexUtils.java:110)
	at io.openliberty.microprofile.openapi20.utils.IndexUtils.indexContainer(IndexUtils.java:122)
	at io.openliberty.microprofile.openapi20.utils.IndexUtils.indexContainer(IndexUtils.java:122)
	at io.openliberty.microprofile.openapi20.utils.IndexUtils.indexContainer(IndexUtils.java:122)
	at io.openliberty.microprofile.openapi20.utils.IndexUtils.indexContainer(IndexUtils.java:122)
	at io.openliberty.microprofile.openapi20.utils.IndexUtils.getIndexView(IndexUtils.java:79)
	at io.openliberty.microprofile.openapi20.ApplicationProcessor.generateModel(ApplicationProcessor.java:297)
	at io.openliberty.microprofile.openapi20.ApplicationProcessor.processWebModule(ApplicationProcessor.java:234)
	at io.openliberty.microprofile.openapi20.ApplicationProcessor.processApplication(ApplicationProcessor.java:132)
	at io.openliberty.microprofile.openapi20.ApplicationRegistry.addApplication(ApplicationRegistry.java:100)
	at io.openliberty.microprofile.openapi20.ApplicationListener.applicationStarting(ApplicationListener.java:45)
	at com.ibm.ws.container.service.state.internal.ApplicationStateManager.fireStarting(ApplicationStateManager.java:51)
	at com.ibm.ws.container.service.state.internal.StateChangeServiceImpl.fireApplicationStarting(StateChangeServiceImpl.java:50)
	at com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase.preDeployApp(SimpleDeployedAppInfoBase.java:547)
	at com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase.installApp(SimpleDeployedAppInfoBase.java:508)
	at com.ibm.ws.app.manager.module.internal.DeployedAppInfoBase.deployApp(DeployedAppInfoBase.java:349)
	at com.ibm.ws.app.manager.war.internal.WARApplicationHandlerImpl.install(WARApplicationHandlerImpl.java:65)
	at com.ibm.ws.app.manager.internal.statemachine.StartAction.execute(StartAction.java:149)
	at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.enterState(ApplicationStateMachineImpl.java:1352)
	at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.run(ApplicationStateMachineImpl.java:897)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:238)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Steps to Reproduce**  
https://github.com/IBM/FHIR/wiki/Setting-up-for-development#running-with-the-liberty-maven-plugin using branch 4.10.2

**Expected behavior**  
This exception should not occur.

**Diagnostic information:**  
 - OpenLiberty Version: 21.0.0.11
 - Affected feature(s) mpOpenApi-2.0 & 3.0
 - Java Version: [i.e. full output of `java -version`]
 ```
openjdk version "11.0.13" 2021-10-19
OpenJDK Runtime Environment 18.9 (build 11.0.13+8)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.13+8, mixed mode, sharing)
```

**Comment by Azquelt** [comment]
From my trace, it looks like the class which was being scanned at the time of the exception was `com.google.common.collect.AbstractSortedSetMultimap` from `guava-31.0.1-jre.jar`

**Comment by Azquelt** [comment]
Looks like this is an issue in jandex which is fixed in 2.4.1: wildfly/jandex#147

**Comment by Azquelt** [comment]
The fix for this problem will be included in the next release (22.0.0.1).
